content/2017.03.building-projects-as-a-way-to-learn.rst

```diff
@@ -0,0 +1,200 @@
+Building software as a way to learn
+###################################
+
+:slug: building-software-as-a-way-to-learn
+:date: 2017-03-15
+:lang: en
+:url: building-software-as-a-way-to-learn
+:summary:
+    Having multiple people working on a big project is not as easy as it seems.
+    Here are some elements to move projects forward as a team.
+
+
+Today, I realized that I've been working for Mozilla for three years
+already.
+
+I've learned a lot, and with the storage team we have been part of a
+number of different projects:
+
+- Firefox Hello: Building a WebRTC signaling service.
+- MSISDN auth: Building a password less BrowserID authentication
+  service using a phone number.
+- Syncto: Browsing Firefox Sync using the Kinto protocol.
+- Kinto: Building synchronization services with backup and sharing capability in mind.
+- RemoteServiceWorker: Building a Firefox Service Worker in the cloud
+- WebPush channels: Building a PubSub on top of WebPush
+
+While most of these projects have been built, deployed in production
+and eventually decommissined, some of them are still actively use by
+Firefox users everyday.
+
+
+How do you keep motivated building things that are thrown away?
+===============================================================
+
+Let's be honest, everything we build: cars, houses, and even softwares
+are meant to be eventually thrown away. That's why recycling is so
+important.
+
+The question is why do we keep building them nevertheless.
+
+Our daily job is to help Mozilla make the web a safe place for users
+and prevent big companies to turn them into advertisement products
+without their consent.
+
+One way of doing so is to build software that empower people to use
+internet while being in charge of their own data.
+
+When users give their data to big companies, they are given them
+leverage. In the meantime these companies are offering a service that
+solve people problems, and they get addicted to it and come back for
+more.
+
+We alone cannot compete with companies that are spending millions of
+dollars building great products to capture users.
+
+However, we can build software that, in one hand, is useful for the
+Mozilla mission and makes us reactive to fix Mozilla issues while
+building Firefox, and in the other hand enpower developers to build
+great software where users choose where they want to store their data.
+
+What matters rather than building software is the vision that this
+software carries.
+
+Writting lines of code is cheap and fast. You can build a prototype in
+two weeks and iterate until you are happy with the result.
+
+Then it might be use by some people and it might get some traction for
+few months or few years and it will be eventually forgotten in a git
+repository somewhere forever.
+
+Writting software to solve problems is not enough, to keep motivated
+on the long term.
```

Comment:
I think this paragraph and the two before it should be combined and rethought. On the one hand, it says writing code is fast and easy, but on the other hand, it says that isn't enough.